pkg/jmxfetch/jmxfetch.go

```diff
@@ -267,3 +285,71 @@ func (j *JMXFetch) Up() (bool, error) {
 	err = process.Signal(syscall.Signal(0))
 	return err == nil, err
 }
+
+func (j *JMXFetch) ConfigureCheck(initConfig integration.Data) error {
+	var initConf checkInitCfg
+
+	// unmarshall init config
+	if err := yaml.Unmarshal(initConfig, &initConf); err != nil {
+		return err
+	}
+
+	if j.JavaBinPath == "" {
+		if initConf.JavaBinPath != "" {
+			j.JavaBinPath = initConf.JavaBinPath
+		}
+	}
+
+	if j.JavaOptions == "" {
+		if initConf.JavaOptions != "" {
+			j.JavaOptions = initConf.JavaOptions
+		}
+	}
+
+	if j.JavaToolsJarPath == "" {
+		if initConf.ToolsJarPath != "" {
+			j.JavaToolsJarPath = initConf.ToolsJarPath
+		}
+	}
+	if j.JavaCustomJarPaths == nil {
+		if initConf.CustomJarPaths != nil {
+			j.JavaCustomJarPaths = initConf.CustomJarPaths
+		}
+	}
+
+	return nil
+}
+
+func (j *JMXFetch) ConfigureInstance(instance integration.Data) error {
```

Comment:
ditto for the doc string, and in terms of naming I'd prefer `ConfigureFromInstance`

Reply:
Done.